tests/rptest/tests/self_test_test.py

```diff
@@ -233,3 +233,66 @@ def test_self_test_cancellable(self):
                 # cancelled before it even had a chance to start, resulting in
                 # a 0 value for duration
                 assert report['duration'] >= 0
+
+    @cluster(num_nodes=3)
+    def test_self_test_unknown_test_type(self):
+        """Assert the self test still runs when an invalid test type is passed.
+           This helps ensure the self test will still work in a cluster
+           with mixed versions of Redpanda."""
+        num_nodes = 3
+
+        self.start_redpanda(num_nodes=num_nodes,
+                            si_settings=SISettings(
+                                test_context=self.test_context,
+                                cloud_storage_enable_remote_read=True,
+                                cloud_storage_enable_remote_write=True))
+
+        #Attempt to run with an unknown test type "pandatest"
+        #and possibly unknown "cloud" test.
+        #The rest of the tests should proceed as normal.
+        request_json = {
+            'tests': [{
+                'type': 'pandatest'
+            }, {
```

Comment:
I don't think it's something you can easily test with this pandatest test type, but what happens if the self-test request is sent to a node that knows about the test type? I.e.

There are two cases to consider:
1. the node receiving the self-test admin request is upgraded and others are not
2. the node receiving the self-test admin request is not upgraded but others are

I think this PR mostly focuses on handling case #1, where we just send the request out to all nodes, and all nodes add to this unknown test list. But what happens for case #2?

Reply:
Good point.

This change doesn't cover for case 2. If the controller is not upgraded, it may not recognize some tests, and the request it sends to the upgraded nodes won't contain those unknown tests.

It may be better to rewrite the structure of `self_test_frontend::start_test()` to accept the original `ss::http::request` dispatched to the leader,  fan that out to each follower node in the cluster, and allow each node in the cluster to do their own JSON processing of the request (may require a new `rpc` type that copies the entire `input_stream<char>* content_stream` from the `ss::http::request`). 

I.e, delay/move the JSON processing for the `self_test_request` out of the `admin_server::self_test_start_handler()` to somewhere in the `self_test_backend`.

What do you think?

Reply:
Ah, after taking another look at this I think deferring the parsing of the JSON is difficult, due to the fact that the controller is responsible for helping set up the `netcheck` test that occurs between nodes in the cluster.

All in all, I'm not sure how far we want to go with supporting this corner case- I think the PR as it stands is an improvement over the current behaviour of throwing 

```
unable to start self test: request POST http://127.0.0.1:9644/v1/debug/self_test/start failed: Bad Request, body: "{\"message\": \"Unknown self_test 'type', valid options are 'disk' or 'network'\", \"code\": 400}"
```

in the aforementioned [case 1](https://github.com/redpanda-data/redpanda/pull/21370#discussion_r1687065210).